lib/ansible/modules/network/cloudengine/ce_vrf_af.py

```diff
@@ -0,0 +1,856 @@
+        if vrf_addr_types:
+            for vrf_addr_type in vrf_addr_types:
+                vrf_af_info = dict()
+                for vrf_addr_type_ele in vrf_addr_type:
+                    if vrf_addr_type_ele.tag in ["vrfName", "afType", "vrfRD"]:
+                        vrf_af_info[vrf_addr_type_ele.tag] = vrf_addr_type_ele.text
+                    if vrf_addr_type_ele.tag == 'vpnTargets':
+                        vrf_af_info["vpnTargets"] = list()
+                        for rtargets in vrf_addr_type_ele:
+                            rt_dict = dict()
+                            for rtarget in rtargets:
+                                if rtarget.tag in ["vrfRTValue", "vrfRTType"]:
+                                    rt_dict[rtarget.tag] = rtarget.text
+                            vrf_af_info["vpnTargets"].append(rt_dict)
+                    if vrf_addr_type_ele.tag == 'exVpnTargets':
+                        vrf_af_info["evpnTargets"] = list()
+                        for rtargets in vrf_addr_type_ele:
+                            rt_dict = dict()
+                            for rtarget in rtargets:
+                                if rtarget.tag in ["vrfRTValue", "vrfRTType"]:
+                                    rt_dict[rtarget.tag] = rtarget.text
+                            vrf_af_info["evpnTargets"].append(rt_dict)
+                self.vrf_af_info["vpnInstAF"].append(vrf_af_info)
+
+    def check_params(self):
+        """Check all input params"""
+        # vrf check
+        if not self.vrf or not self.vrf_aftype:
+            self.module.fail_json(
+                msg='Error: The vrf name and address family must be set.')
+        # vrf and description check
+        if self.vrf == '_public_':
+            self.module.fail_json(
+                msg='Error: The vrf name _public_ is reserved.')
+        if not self.get_vrf():
+            self.module.fail_json(
+                msg='Error: The vrf name do not exist.')
+        if self.state == 'present':
+            if self.route_distinguisher:
+                if not is_valid_value(self.route_distinguisher):
+                    self.module.fail_json(msg='Error:The vrf route distinguisher length must between 3 ~ 21,'
+                                          'i.e. X.X.X.X:number<0-65535> or number<0-65535>:number<0-4294967295>'
+                                          'or number<0-65535>.number<0-65535>:number<0-65535>'
+                                          'or number<65536-4294967295>:number<0-65535>'
+                                          ' but not be 0:0 or 0.0:0.')
+            if not self.vpn_target_state:
+                if self.vpn_target_value or self.vpn_target_type:
+                    self.module.fail_json(
+                        msg='Error: The vpn target state should be exist.')
+            if self.vpn_target_state:
+                if not self.vpn_target_value or not self.vpn_target_type:
```

Comment:
There are some options to `AnsibleModule` which you may be able to use to validate options passed in.
Look at existing modules for examples:
```
mutually_exclusive
required_together
required_one_of
require_if
```